truffleruby/src/main/java/org/truffleruby/core/thread/ThreadManager.java

```diff
@@ -391,4 +484,15 @@ public synchronized void accept(DynamicObject thread, Node currentNode) {
         }
     }
 
+    @TruffleBoundary
+    public void interrupt(Thread thread) {
+        final UnblockingAction action = unblockingActions.get(thread);
```

Comment:
It's bound by the number of alive threads.
When a thread dies, `cleanupValuesBasedOnCurrentJavaThread()` is called which removes the entry from the map.
Also, the unblocking action is only in the map for the duration of the blocking call, after it is restored to EMPTY_UNBLOCKING_ACTION (because we cannot use `null` for a ConcurrentHashMap).